libs/storage/map_files_downloader.cpp

```diff
@@ -44,12 +44,12 @@ void MapFilesDownloader::RunMetaConfigAsync(std::function<void()> && callback)
 
   GetPlatform().RunTask(Platform::Thread::Network, [this, callback = std::move(callback)]()
   {
-    GetMetaConfig([this, callback = std::move(callback)](MetaConfig const & metaConfig)
+    GetMetaConfig([this, callback = std::move(callback)](downloader::MetaConfig && metaConfig)
     {
-      m_serversList = metaConfig.m_serversList;
-      settings::Update(metaConfig.m_settings);
-      products::Update(metaConfig.m_productsConfig);
-      callback();
+      m_serversList = std::move(metaConfig.servers);
+      settings::Update(metaConfig.settings);
+      products::ProductsSettings::Instance().Update(std::move(metaConfig.productsConfig));
+      GetPlatform().RunTask(Platform::Thread::Gui, std::move(callback));
```

Comment:
I'd prefer to revert the old behaviour with SafeCallback:
- before, all stuff (including assign m_serversList = ) happens happens in Gui thread (safe)
- now, only callback is called in Gui thread.

Reply:
@vng fixed, PTAL